generate_figures/deep_sets_linear_profiling_var_n.py

```diff
@@ -2,103 +2,74 @@
-quant_test_u_rel_error_deep.reset_index(inplace=True)
-quant_test_u_rel_error_deep.columns = ['N'] + [f'quantile_{q}' for q in quantiles]
-
-plt.rcParams.update(params) 
-
-
-ax_time = plt.subplot(121)
-plt.plot(quant_train_time_deep['N'], quant_train_time_deep['quantile_0.5'], label= r"Median")
-plt.fill_between(quant_train_time_deep['N'], quant_train_time_deep["quantile_0.1"],quant_train_time_deep["quantile_0.9"], color ='cornflowerblue', alpha=0.2, label = r"$10$th and $90$th percentiles")
-plt.fill_between(quant_train_time_deep['N'], quant_train_time_deep["quantile_0.25"],quant_train_time_deep["quantile_0.75"], color ='cornflowerblue', alpha=0.6, label = r"$25$th and $75$th percentiles")
-ax_time.set_xscale('log')
-ax_time.xaxis.set_ticks([50, 100, 1000, 10000, 100000])
-plt.title(r"Computation time (seconds)")
-plt.xlabel(r"N")
-plt.legend(prop={"size": fontsize}, loc='upper left')
-plt.tight_layout()
-
-
-
-ax_loss = plt.subplot(122)
-plt.plot(quant_test_u_rel_error_deep['N'], quant_test_u_rel_error_deep['quantile_0.5'], label=r"Median")
-plt.fill_between(quant_test_u_rel_error_deep['N'], quant_test_u_rel_error_deep["quantile_0.1"],quant_test_u_rel_error_deep["quantile_0.9"], color ='cornflowerblue', alpha=0.2, label = r"$10$th and $90$th percentiles")
-plt.fill_between(quant_test_u_rel_error_deep['N'], quant_test_u_rel_error_deep["quantile_0.25"],quant_test_u_rel_error_deep["quantile_0.75"], color ='cornflowerblue', alpha=0.6, label = r"$25$th and $75$th percentiles")
-ax_loss.set_xscale('log')
-ax_loss.set_yscale('log')
-ax_loss.xaxis.set_ticks([50, 100, 1000, 10000, 100000])
-ax_loss.yaxis.set_ticks([0.0001, 0.001]) 
-plt.title(r"Policy errors ($\epsilon_{\mathrm{rel}}$)")
-plt.xlabel(r"N")
-plt.legend(prop={"size": fontsize}, loc='lower left')
-plt.tight_layout()
 
-plt.savefig(output_path)
+for n, vals in plot_dict.items():
+    plt.rcParams.update(params)
+    df = df_deep.groupby("N")[n].quantile(quantiles).unstack(level=-1)
+    df.reset_index(inplace=True)
+    df.columns = ["N"] + [f"quantile_{q}" for q in quantiles]
+    ax = plt.subplot(vals["subplot"])
+    plt.plot(df["N"], df["quantile_0.5"], label=r"Median")
+    plt.fill_between(
+        df["N"],
+        df["quantile_0.1"],
+        df["quantile_0.9"],
+        color="cornflowerblue",
+        alpha=0.2,
+        label=r"$10$th and $90$th percentiles",
+    )
+    plt.fill_between(
+        df["N"],
+        df["quantile_0.25"],
+        df["quantile_0.75"],
+        color="cornflowerblue",
+        alpha=0.6,
+        label=r"$25$th and $75$th percentiles",
+    )
+    ax.set_xscale("log")
+    if n == "test_u_rel_error":
+        ax.set_yscale("log")
+
+    ax.xaxis.set_ticks(vals["ticks_x"])
+    ax.yaxis.set_ticks(vals["ticks_y"])
+    plt.title(vals["title"])
+    plt.xlabel(r"N")
+    plt.legend(prop={"size": params["font.size"]}, loc=vals["loc"])
+    plt.tight_layout()
+plt.show()
```

Comment:
All the plots and tables have just `plt.show()` instead of `plt.savefig()` or` \..figures` because I figure there will be more testing and its easier to do in the interactive window for me but I will change it to the correct one later.